pkg/sql/codegen/ir_to_proto.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codegen
+
+import (
+	"fmt"
+	"strings"
+
+	pb "sqlflow.org/sqlflow/pkg/server/proto"
+	irpb "sqlflow.org/sqlflow/pkg/sql/codegen/proto"
+)
+
+// FIXME(typhoonzero): copied from tensorflow/codegen.go
+func intArrayToJSONString(ia []int) string {
+	return strings.Join(strings.Split(fmt.Sprint(ia), " "), ",")
+}
+
+// FIXME(typhoonzero): copied from tensorflow/codegen.go
+func attrToPythonValue(attr interface{}) string {
+	switch attr.(type) {
+	case int:
+		return fmt.Sprintf("%d", attr.(int))
+	case int64:
+		return fmt.Sprintf("%d", attr.(int64))
+	case float32:
+		return fmt.Sprintf("%f", attr.(float32))
+	case float64: // FIXME(typhoonzero): may never use
+		return fmt.Sprintf("%f", attr.(float64))
+	case []int:
+		return intArrayToJSONString(attr.([]int))
+		// TODO(typhoonzero): support []float etc.
+	case []interface{}:
+		tmplist := attr.([]interface{})
+		if len(tmplist) > 0 {
+			if _, ok := tmplist[0].(int); ok {
+				intlist := []int{}
+				for _, v := range tmplist {
+					intlist = append(intlist, v.(int))
+				}
+				return intArrayToJSONString(intlist)
+			}
+		}
+		// TODO(typhoonzero): support []float etc.
+		return "[]"
+	case string:
+		return attr.(string)
+	default:
+		return ""
+	}
+}
+
+// FIXME(typhoonzero): copied from tensorflow/codegen.go
+func dtypeToString(dt FieldType) string {
+	switch dt {
+	case Float:
+		return "float32"
+	case Int:
+		return "int64"
+	case String:
+		return "string"
+	default:
+		return ""
+	}
+}
+
+func toInt32List(il []int) []int32 {
+	ret := []int32{}
+	for _, i := range il {
+		ret = append(ret, int32(i))
+	}
+	return ret
+}
+
+func fieldMetaToPbMeta(fm *FieldMeta) *irpb.FieldMeta {
```

Comment:
Naming: `Pb` or `PB`? I feel `PB` is better since it is the acronym of Protocol Buffers.